modules/caddyhttp/reverseproxy/reverseproxy.go

```diff
@@ -1446,7 +1446,8 @@ func (c ignoreClientGoneContext) Err() error {
 // from the proxy handler.
 const proxyHandleResponseContextCtxKey caddy.CtxKey = "reverse_proxy_handle_response_context"
 
-var noUpstreamsAvailable = fmt.Errorf("no upstreams available")
+// ErrNoUpstream occurs when there are no upstream available.
+var ErrNoUpstream = fmt.Errorf("no upstreams available")
```

Comment:
Why does this have to be exported?

Reply:
That allows users to unwrap errors using 
```
errors.Is(err, reverse.ErrNoUpstream)
```
but I might be missing something, so maybe it doesn't. 
What it does need to do is start with either `err` or `Err` to pass linting.

Reply:
```suggestion
// errNoUpstream occurs when there are no upstream available.
var errNoUpstream = fmt.Errorf("no upstreams available")
```

If it's OK with you I'm going to leave it unexported for now, until we have a request for this, as I don't want to expose more API surface than I have to.

If this LGTY, then I'm ready to approve and merge this PR :smiley:

Reply:
@AnomalRoil can you adjust the above to not export this error? I think we can merge after that.

Reply:
Sorry I had missed these comments in my GH notifications. Done.